### PR TITLE
NE-553: Implement the PROXY protocol API

### PR DESCRIPTION
**Pull request by Miciah**
#### Bump openshift/api for the PROXY protocol API

Bump to github.com/openshift/api@e37aaed4c2784348bd19a794dcfe7f63e26000d2 to get the PROXY protocol API.

* `go.mod`: Bump.
* `go.sum`:
* `manifests/00-custom-resource-definition.yaml`:
* `pkg/manifests/bindata.go`:
* `vendor/github.com/openshift/api/*`:
* `vendor/modules.txt`: Regenerate.

#### `test/e2e`: Rename `nondefaultAvailableConditions`

* `test/e2e/operator_test.go` (`nondefaultAvailableConditions`): Rename to `availableConditionsForIngressControllerWithLoadBalancer`.

#### Implement the PROXY protocol API

This resolves [NE-553](https://issues.redhat.com/browse/NE-553).

* `pkg/operator/controller/ingress/controller.go` (`setDefaultPublishingStrategy`): Set protocol for the `HostNetwork` and `NodePortService` endpoint publishing strategy types.
(`IsProxyProtocolNeeded`): Check `status.endpointPublishingStrategy.hostNetwork.protocol` or `status.endpointPublishingStrategy.hostNetwork.protocol` as appropriate and return true if the value is "PROXY".
* `pkg/operator/controller/ingress/controller_test.go` (`TestIsProxyProtocolNeeded`): New test.  Verify that `IsProxyProtocolNeeded` behaves as expected for various platforms and endpoint publishing strategy parameters.
* `test/e2e/operator_test.go` (`availableConditionsForIngressControllerWithNodePort`): New variable.
(`TestClusterProxyProtocol`): Rename...
(`TestProxyProtocolOnAWS`): ...to this.
(`TestProxyProtocolAPI`): New test.  Verify that the operator properly configures PROXY protocol when it is specified with the `NodePortService` endpoint publishing strategy.
(`waitForDeploymentEnvVar`): Treat unset environment variable and environment variable with empty value as equivalent.

### CODE REVIEW

File: test/e2e/operator_test.go [hunk 1]
```diff
@@ -230,6 +239,64 @@ func TestClusterProxyProtocol(t *testing.T) {
...
+		t.Fatalf("expected initial deployment not to enable PROXY protocol")
+	}
+
+	ic.Spec.EndpointPublishingStrategy.NodePort = &operatorv1.NodePortStrategy{
+		Protocol: operatorv1.ProxyProtocol,
+	}
+	if err := kclient.Update(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+	err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
```
Comment (sgreene570) [inline test/e2e/operator_test.go:283]: Could you reuse ` waitForDeploymentEnvVar` here? 

https://github.com/openshift/cluster-ingress-operator/blob/master/test/e2e/operator_test.go#L1864

File: test/e2e/operator_test.go [hunk 2]
```diff
@@ -1860,7 +1906,9 @@ func waitForDeploymentComplete(t *testing.T, cl client.Client, deployment *appsv
 	return nil
 }
 
-// Wait for the provided deployment to have the specified environment variable set.
+// Wait for the provided deployment to have the specified environment variable
```
Comment (sgreene570) [inline test/e2e/operator_test.go:1909]: Nice fixup!

**Comment by sgreene570** [comment]
Looks great! Would be happy to LGTM once this has been rebased :smile:

**Comment by Miciah** [comment]
Rebased.

**Comment by sgreene570** [comment]
/lgtm
:ship:

**Pull request merged**